jinahub/rankers/MinRanker/minranker.py

```diff
@@ -0,0 +1,56 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from itertools import groupby
+from typing import List, Dict
+
+from jina import Executor, requests, DocumentArray
+from jina.logging.logger import JinaLogger
+
+
+class MinRanker(Executor):
+    """
+    :class:`MinRanker` aggregates the score of the matched doc from the matched chunks.
+    For each matched doc, the score is aggregated from all the matched chunks belonging to that doc.
+    
+    :param metric: the distance metric used in `scores`
+    :param default_traversal_paths: traverse path on docs, e.g. ['r'], ['c']
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        metric: str = None,
+        default_traversal_paths: List[str] = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.logger = JinaLogger(self.__class__.__name__)
+        if not metric:
+            self.logger.error("metric should not be None, please set it the same as your indexer's metric")
+        self.metric = metric
+        self.default_traversal_paths = default_traversal_paths or ['r']
```

Comment:
This is also not needed - simply adjust
```python
default_traversal_paths: List[str] = ['r'],
```
in function definition

Reply:
Actually, the best version would be
```python
default_traversal_paths: Iterable[str] = ('r'),
```

Reply:
Ok, I see. The validation is not so straightforward due to the current design of scores. I will adopt your suggestion, thanks!